web/src/components/storage/iscsi/AuthFields.jsx

```diff
@@ -0,0 +1,157 @@
+  const onPasswordChange = v => onChange("password", v);
+  const onReverseUsernameChange = v => onChange("reverseUsername", v);
+  const onReversePasswordChange = v => onChange("reversePassword", v);
+
+  const isValidText = v => v.length > 0;
+  const isValidUsername = () => isValidText(data.username);
+  const isValidPassword = () => isValidText(data.password);
+  const isValidReverseUsername = () => isValidText(data.reverseUsername);
+  const isValidReversePassword = () => isValidText(data.reversePassword);
+  const isValidAuth = () => isValidUsername() && isValidPassword();
+
+  const showUsernameError = () => isValidPassword() ? !isValidUsername() : false;
+  const showPasswordError = () => isValidUsername() ? !isValidPassword() : false;
+  const showReverseUsernameError = () => {
+    return (isValidAuth() && isValidReversePassword()) ? !isValidReverseUsername() : false;
+  };
+  const showReversePasswordError = () => {
+    return (isValidAuth() && isValidReverseUsername()) ? !isValidReversePassword() : false;
+  };
+
+  useEffect(() => {
+    onValidate(
+      !showUsernameError() &&
+      !showPasswordError() &&
+      !showReverseUsernameError() &&
+      !showReversePasswordError()
+    );
+  });
+
+  const ByInitiatorAuthLegend = () => {
+    const Info = () => {
+      return (
+        <p>
+          <Icon
+            name="info"
+            size="16"
+            style={{ verticalAlign: "text-bottom", marginRight: "0.3em" }}
+          />
+          Only available if authentication by target is provided
+        </p>
+      );
+    };
+
+    return (
+      <>
+        Authentication by initiator
+        { !isValidAuth() && <Info /> }
+      </>
+    );
+  };
+
+  return (
+    <>
+      <Fieldset legend="Authentication by target">
+        <FormGroup
+          fieldId="username"
+          label="Username"
+          helperTextInvalid="Incorrect username"
+          validated={showUsernameError() ? "error" : "default"}
+        >
+          <TextInput
+            id="username"
+            name="username"
+            aria-label="Username"
+            value={data.username || ""}
+            label="Username"
+            onChange={onUsernameChange}
+            validated={showUsernameError() ? "error" : "default"}
+          />
+        </FormGroup>
+        <FormGroup
+          fieldId="password"
+          label="Password"
+          helperTextInvalid="Incorrect password"
+          validated={showPasswordError() ? "error" : "default"}
+        >
+          <TextInput
+            id="password"
+            name="password"
+            type="password"
+            aria-label="Password"
+            value={data.password || ""}
+            label="Password"
+            onChange={onPasswordChange}
+            validated={showPasswordError() ? "error" : "default"}
+          />
+        </FormGroup>
+      </Fieldset>
+      <Fieldset legend={ByInitiatorAuthLegend()}>
```

Comment:
NP: Not sure if it's a good idea adding the additional warning as part of the legend. But keep it as it is now, we can refine it in a later iteration.